src/util/schnorr.rs

```diff
@@ -17,3 +17,64 @@
 //!
 
 pub use secp256k1::schnorrsig::{PublicKey, KeyPair};
+use secp256k1::{Secp256k1, Verification};
+use hashes::{Hash, HashEngine};
+use util::taproot::{TapBranchHash, TapTweakHash};
+
+/// Untweaked Schnorr public key
+pub struct UntweakedPublicKey(pub PublicKey);
+
+/// Tweaked Schnorr public key
+pub struct TweakedPublicKey(PublicKey);
+
+impl UntweakedPublicKey {
+    /// Tweaks an untweaked public key given an untweaked key and optional script tree merkle root.
+    /// 
+    /// This is done by using the equation Q = P + H(P|c)G, where 
+    ///  * Q is the tweaked key
+    ///  * P is the internal key
+    ///  * H is the hash function
+    ///  * c is the commitment data
+    ///  * G is the generator point
+    pub fn tweak<C: Verification>(&self, secp: Secp256k1<C>, merkle_root: Option<TapBranchHash>) -> TweakedPublicKey {
```

Comment:
propose to rename the method into `tap_tweak` (since there may be other tweaks, like P2C)